docs/sketch-specification.md

```diff
@@ -35,7 +35,7 @@ The following extensions are supported:
 - .cpp - C++ files.
 - .c - C Files.
 - .S - Assembly language files.
-- .h - Header files.
+- .h, .hpp, .hh - Header files.
 - .tpp, .ipp - Header files (available from Arduino CLI 0.19.0).
 
```

Comment:
```suggestion
- .h, .hpp, .hh [<sup>1</sup>](#hpp-hh-note) - Header files.
- .tpp, .ipp [<sup>2</sup>](#tpp-ipp-note) - Header files.

<a id="hpp-hh-note"></a> <sup>1</sup> `.hpp` and `.hh` supported from Arduino IDE 1.8.0/arduino-builder 1.3.22. <br />
<a id="tpp-ipp-note"></a> <sup>2</sup> Supported from Arduino CLI 0.19.0.

```

References:

- https://github.com/arduino/arduino-builder/commit/1c1369d3b19351b1aef42d4c08cef211dbf4c9aa
- https://github.com/arduino/Arduino/commit/df0009601ef8747a307241fb4c3f1b94d9f96127

The need to specify that the note only applies to `.hpp` and `.hh` made the length of the text exceed what would be appropriate to add in the list IMO. So I used the "footnote" style as established here:

https://arduino.github.io/arduino-cli/dev/library-specification/#version-constraints

For the sake of consistency, I also moved the less verbose note re the `.tpp` and `.ipp` extensions to a footnote.